### ISSUE TITLE
[Have workaround, definitely a bug] Cannot projects.get multiple projects in same request (wrong format)

### ISSUE THREAD

**Issue by thehappycoder** [XeroAPI/xero-node#302]
I am using xero-node@3.1.x

To reproduce:

```
const projectsAPIClient = new ProjectsAPIClient({
    appType: 'private',
    consumerKey: env.XERO_CONSUMER_KEY!,
    consumerSecret: env.XERO_CONSUMER_SECRET!,
    privateKeyString: env.XERO_PRIVATE_KEY_STRING!
  })

await projectsAPIClient.projects.get({
    projectIds: [
    '229c6fd7-f157-4234-bb48-ab14d4674035',
    '229c6fd7-f157-4234-bb48-ab14d4674036'
  ]
```

This generates the following query string: `projects?projectIds=229c6fd7-f157-4234-bb48-ab14d4674035&projectIds=229c6fd7-f157-4234-bb48-ab14d4674036`

which fails:
```
[5/12/19 4:13:23 am] { Error: XeroError: signature_invalid (Failed to validate signature)
[5/12/19 4:13:23 am]     at oauthLib.get (/Users/anton/projects/cms-xero-integration/node_modules/xero-node/lib/internals/OAuth1HttpClient.js:233:49)
[5/12/19 4:13:23 am]     at passBackControl (/Users/anton/projects/cms-xero-integration/node_modules/oauth/lib/oauth.js:397:13)
[5/12/19 4:13:23 am]     at IncomingMessage.<anonymous> (/Users/anton/projects/cms-xero-integration/node_modules/oauth/lib/oauth.js:409:9)
[5/12/19 4:13:23 am]     at IncomingMessage.emit (events.js:203:15)
[5/12/19 4:13:23 am]     at endReadableNT (_stream_readable.js:1145:12)
[5/12/19 4:13:23 am]     at process._tickCallback (internal/process/next_tick.js:63:19)
[5/12/19 4:13:23 am]   statusCode: 401,
[5/12/19 4:13:23 am]   data:
[5/12/19 4:13:23 am]    'oauth_problem=signature_invalid&oauth_problem_advice=Failed%20to%20validate%20signature',
[5/12/19 4:13:23 am]   headers:
[5/12/19 4:13:23 am]    { 'content-type': 'text/html; charset=utf-8',
[5/12/19 4:13:23 am]      'strict-transport-security': 'max-age=31536000',
[5/12/19 4:13:23 am]      'www-authenticate': 'OAuth Realm="api.xero.com"',
[5/12/19 4:13:23 am]      'content-length': '87',
[5/12/19 4:13:23 am]      expires: 'Thu, 05 Dec 2019 04:13:23 GMT',
[5/12/19 4:13:23 am]      'cache-control': 'max-age=0, no-cache, no-store',
[5/12/19 4:13:23 am]      pragma: 'no-cache',
[5/12/19 4:13:23 am]      date: 'Thu, 05 Dec 2019 04:13:23 GMT',
[5/12/19 4:13:23 am]      connection: 'close',
[5/12/19 4:13:23 am]      'x-client-tls-ver': 'tls1.2' } }
[5/12/19 4:13:23 am] Executed 'Functions.cmsToXero' (Failed, Id=0b5746c8-0ab4-45fb-9ef2-5d63b18c7352)
[5/12/19 4:13:23 am] System.Private.CoreLib: Exception while executing function: Functions.cmsToXero. System.Private.CoreLib: Result: Failure
[5/12/19 4:13:23 am] Exception: Error: XeroError: signature_invalid (Failed to validate signature)
[5/12/19 4:13:23 am] Stack: Error: XeroError: signature_invalid (Failed to validate signature)
[5/12/19 4:13:23 am]     at oauthLib.get (/Users/anton/projects/cms-xero-integration/node_modules/xero-node/lib/internals/OAuth1HttpClient.js:233:49)
[5/12/19 4:13:23 am]     at passBackControl (/Users/anton/projects/cms-xero-integration/node_modules/oauth/lib/oauth.js:397:13)
[5/12/19 4:13:23 am]     at IncomingMessage.<anonymous> (/Users/anton/projects/cms-xero-integration/node_modules/oauth/lib/oauth.js:409:9)
[5/12/19 4:13:23 am]     at IncomingMessage.emit (events.js:203:15)
[5/12/19 4:13:23 am]     at endReadableNT (_stream_readable.js:1145:12)
[5/12/19 4:13:23 am]     at process._tickCallback (internal/process/next_tick.js:63:19).
```

When I checked the documentation https://developer.xero.com/documentation/projects/projects, it has an example where the query string is in different format: `https://.../projects?projectIDs={projectId},{projectId}`.

When I cowboy the `node_modules/xero-node/lib/ProjectsAPIClient.js` and set the endpoint to `endpoint = 'projects?projectIDs=229c6fd7-f157-4234-bb48-ab14d4674035,229c6fd7-f157-4234-bb48-ab14d4674036'`, it works well.

The workaround is to pass a string instead of an array (have to cast to any because typescript interface expects an array):
```
await projectsAPIClient.projects.get({
    projectIds: projectIds.join(',') as any
  })
```

**Comment by thehappycoder** [comment]
PR: https://github.com/XeroAPI/xero-node/pull/303

**Comment by SerKnight** [comment]
Thanks for this catch @thehappycoder  - we are working out the kinks in the types that are generated from the OpenAPI spec. Glad you found the interim fix and shared this, will be re-integrating solve asap.

**Comment by RettBehrens** [comment]
A quick note on why we’re closing this issue:

With the release of version 4.x of the Xero-Node SDK in December of 2019, we delivered a complete overhaul of the SDK. 

We moved from writing our SDKs by hand to generating our SDKs using our [Xero-OpenAPI](https://github.com/XeroAPI/Xero-OpenAPI) specs. By utilizing Xero-OpenAPI specs to generate code we’ve improved the developer experience of maintaining the SDK and improved the accessibility for our developer community to contribute via pull requests to the Xero-OpenAPI specs. 

We also built our new SDKs to use the industry standard OAuth2 in preparation for our March 2021 deprecation of OAuth1. For more details, see our [blog post on our decision to move to OAuth2](https://devblog.xero.com/an-update-on-why-we-are-saying-goodbye-oauth-1-0a-hello-oauth-2-0-6a839230908f). 

Version 3.x using Oauth1 and version 4.x using OAuth2 do not share any code as we now generate our SDKS. Going forward we’re focusing on building, maintaining and improving our generated SDKs but you are welcome to fork the [‘oauth1’ branch](https://github.com/XeroAPI/xero-node/tree/oauth1) and apply changes if you wish to implement something for your integration.
